packages/sif-common-formik-ds/src/validation/normalizeNumberString.ts

```diff
@@ -0,0 +1,105 @@
+    const hasDot = cleanedValue.includes('.');
+    const hasSpace = cleanedValue.includes(' ');
+
+    if (hasComma && hasDot) {
+        // Determine which character is the decimal separator
+        const lastCommaIndex = cleanedValue.lastIndexOf(',');
+        const lastDotIndex = cleanedValue.lastIndexOf('.');
+        const decimalSeparator = lastCommaIndex > lastDotIndex ? ',' : '.';
+        const thousandSeparator = decimalSeparator === ',' ? '.' : ',';
+        const integerValue = cleanedValue.split(decimalSeparator)[0];
+
+        if (hasSpace) {
+            if (!verifyThousandsSeparatorIsValid(integerValue, ' ')) {
+                throw new Error(INVALID_NUMBER_FORMAT);
+            }
+            cleanedValue = cleanedValue.replace(/\s+/g, '');
+        }
+
+        // Verify that all the thousands separators are in the right place
+        if (!verifyThousandsSeparatorIsValid(integerValue, thousandSeparator)) {
+            throw new Error(INVALID_NUMBER_FORMAT);
+        }
+
+        // Replace the thousands separators and convert to number
+        const normalizedValue = cleanedValue
+            .replace(new RegExp(`\\${thousandSeparator}`, 'g'), '')
+            .replace(decimalSeparator, '.');
+
+        return normalizedValue;
+    }
+
+    // If only one separator is present
+    if (hasComma || hasDot) {
+        const separator = hasComma ? ',' : '.';
+        const parts = cleanedValue.split(separator);
+
+        // If space is present, and parts length === 2 it's probably a thousands separator - remove it
+        cleanedValue =
+            parts.length === 2 && cleanedValue.indexOf(' ') < cleanedValue.indexOf(separator)
+                ? cleanedValue.replace(/\s+/g, '')
+                : cleanedValue;
+
+        if (cleanedValue.includes(' ')) {
+            throw new Error(INVALID_NUMBER_FORMAT);
+        }
+
+        // More than one separator of the same type, it's probaly a thousands separator
+        if (parts.length > 2) {
+            if (!verifyThousandsSeparatorIsValid(cleanedValue, separator)) {
+                throw new Error(INVALID_NUMBER_FORMAT);
+            }
+            const normalizedValue = cleanedValue.replace(new RegExp(`\\${separator}`, 'g'), '');
+            return normalizedValue;
+        }
+
+        // Only one separator of the same type
+        const fractionalPart = parts[parts.length - 1];
+
+        // It's a decimal separator since the fractional part has 2 or less digits,
+        // or more than 3 digits
+        // 100.0, 100.00, 100.1231
+        if (fractionalPart.length <= 2 || fractionalPart.length > 3) {
+            const normalizedValue = cleanedValue.replace(separator, '.');
+            return normalizedValue;
+        }
+        // When the deciaml has 3 digits is indecisive if it's a decimal
+        // separator or a thousands separator
+        throw new Error(INVALID_NUMBER_FORMAT);
+    }
+
+    // Check if space is used as thoursands separator
```

Comment:
```suggestion
    // Check if space is used as thousands separator
```

<img src="https://media0.giphy.com/media/wlnicjV6gPKodha404/giphy-downsized-medium.gif"/>